svc-events/events/evtsubscription.go

```diff
@@ -1066,12 +1057,20 @@ func (p *PluginContact) createEventSubscrption(taskID string, subTaskChan chan<-
 	var (
 		subTaskURI      string
 		subTaskID       string
+		reqBody         []byte
+		reqJSON         string
 		err             error
 		resp            errResponse.RPC
 		percentComplete int32
 	)
 	defer wg.Done()
 
+	reqBody, err = json.Marshal(request)
+	if err != nil {
+		log.Println("error while trying to marshal create event request: ", err.Error())
```

Comment:
Shouldn't this error be returned?

Reply:
The function doesn't have a return type. Plus as input to the Marshal function here is a well-defined struct, so there are no chances for an error to occur. Even if an error occurs, the flow won't be affected and the only difference will be JSONBody in the payload will have an empty string.

Reply:
If an error is being captured then it should be handled.

Reply:
Error handling for the entire function has to be handled. This will be taken up as a back log item.